src/rate_limiter.cc

```diff
@@ -319,6 +317,19 @@ RateLimiter::GetPayload(
 void
 RateLimiter::PayloadRelease(std::shared_ptr<Payload>& payload)
 {
+  // If this is an exit payload, the instance must not be staged once marked as
+  // available, so mark the instance as removing.
```

Comment:
Just for my understanding, this is done after the response has been sent?

Reply:
No, this is called immediately after the inference request(s) inside the payload is sent to the backend for inferencing, so the response is likely not yet available when this happens.